cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -607,6 +586,46 @@ public void onBecomeFollowerFromOffline(Message message, NotificationContext con
       }
     }
 
+    // rebuild the replica if there is at least one live replica to rebuild from AND it meets either of TWO conditions:
+    // Case 1. DB local meta is different from upstream
+    // Case 2. DB local meta is the same as upstream, but:
+    //   - latest local data update is too old (WAL TTL expired)
+    //   - sequence number is not the same as upstream (otherwise no need to rebuild)
```

Comment:
In case2, it is actually not very accurate to say "what condition we will rebuild", it is actually "what condition we will skip rebuild". 
- for those conditions, we will skip, but. Logically, this equals to "only rebuild when both are true". But, it is actually quite hard to understand why we only rebuild when both are true. 

I think the wording of "case2" are confusing.

Reply:
updated, let me know if it's better